File: aerox.Runtime.Audio/AudioModule.cs
using ManagedBass;

namespace aerox.Runtime.Audio;

[NativeRuntimeModule]
public class AudioModule : RuntimeModule
{
    public override void Startup(Runtime runtime)
    {
        base.Startup(runtime);
        Bass.Init();
    }

    public override void Shutdown(Runtime runtime)
    {
        base.Shutdown(runtime);
        Bass.Free();
    }
}

File: modules/audio/lib/aerox/audio/AudioModule.cpp
#include "aerox/core/GRuntime.hpp"
#include "aerox/core/utils.hpp"
#include "aerox/audio/AudioModule.hpp"
#include "bass/Bass.hpp"
namespace aerox::audio {
    std::string AudioModule::GetName()
    {
        return "Audio Module";
    }

    void AudioModule::Startup(GRuntime* runtime)
    {
        bass::init(-1,44100,0,nullptr);
        bass::setConfig(bass::Config::GVolStream,2500.0f);
    }

    void AudioModule::Shutdown(GRuntime* runtime)
    {
        bass::stop();
    }

    bool AudioModule::IsDependentOn(Module* module)
    {
        return false;
    }
}
